### ISSUE TITLE
inserting literal block scalar - Lexer error

### ISSUE THREAD

**Issue by cswarth** [mikefarah/yq#628]
I'm sorry to bother you with this, but I can't seem to make my literal block scalar work in yq version 4.0.0 and I don't know if it's a bug or just my bad syntax.

I'm trying to turn this, 
```
mapUsers: |
  []

```
into this, 

```
  mapUsers: |
    - userarn: arn:aws:iam::12345678:root
      username: admin
      groups:
        - system:masters

```
In fact I can't seem to insert any block literal of more than one word.
```
$ yq --version
yq version 4.0.0

$ cat auth.yaml
mapUsers: |
  []

$ yq eval '.mapUsers="foo bar"' auth.yaml
Error: Parsing expression: Lexer error: could not match text starting at 1:11 failing at 1:16.
	unmatched text: "\"foo "
I've tried various recipes but I keep getting this error, which looks like it is trying to parse the string I am inserting.

```

Could I see an example of inserting a multiline literal block scalar?

**Comment by mikefarah** [comment]
Oh there was a bug with spaces and scalars that a new version fixed (4.2.0) try that.

BTW this example looks strange - are you encoding a yaml document with a yaml document as a string?

**Comment by cswarth** [comment]
I agree, it does look strange!  
but that is the format created by AWS for managing users in EKS clusters
https://docs.aws.amazon.com/eks/latest/userguide/add-user-role.html

```
apiVersion: v1
data:
  mapRoles: |
    - rolearn: <arn:aws:iam::111122223333:role/eksctl-my-cluster-nodegroup-standard-wo-NodeInstanceRole-1WP3NUE3O6UCF>
      username: <system:node:{{EC2PrivateDNSName}}>
      groups:
        - <system:bootstrappers>
        - <system:nodes>
  mapUsers: |
    - userarn: <arn:aws:iam::111122223333:user/admin>
      username: <admin>
      groups:
        - <system:masters>
    - userarn: <arn:aws:iam::111122223333:user/ops-user>
      username: <ops-user>
      groups:
        - <system:masters>
```

**Comment by cswarth** [comment]
yq  4.2.0 doesn't give the same error message, but the output doesn't look right, does it?
Is  "|2"  legal syntax?
 
```
$ yq --version
yq version 4.2.0

$ cat auth.yaml
mapUsers: |
  []

$ yq eval '.mapUsers="
- foo
- bar
"' auth.yaml
mapUsers: |2

  - foo
  - bar

```

**Comment by cswarth** [comment]
Fixed in yq 4.2.0
